commit create crud book

krcg93 opened the pull request:
## Summary
- Completed book crud and custom exceptions 

## Trello Card
https://trello.com/c/IotStoeI/18-crear-las-operaciones-crud-de-book

krcg93 committed:
commit create crud book

krcg93 committed:
add comments to all methods controller

juandbc commented on src/main/java/wolox/training/exceptions/BookNotFoundException.java:
```diff
@@ -0,0 +1,16 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.FORBIDDEN)
```
```suggestion
@ResponseStatus(HttpStatus.NOT_FOUND)
```

juandbc commented on src/main/java/wolox/training/exceptions/BookIdMismatchException.java:
```diff
@@ -0,0 +1,7 @@
+package wolox.training.exceptions;
+
+public class BookIdMismatchException extends RuntimeException{
```
Here `@ResponseStatus` is missing

juandbc replied:
In this case, ** BAD REQUEST ** will be the best option, because the request data is incorrect.

juandbc commented on src/main/java/wolox/training/controllers/BookController.java:
```diff
@@ -1,17 +1,57 @@
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/author/{author}")
```
Here the idea is to get the author as query param and not as path variable

krcg93 committed:
http status is changed, the route is removed and the parameter is left, the type of data received as a parameter is changed

krcg93 committed:
Merge branch 'feature-bookcrud' into feature-javadoc

krcg93 committed:
Merge pull request #6 from wolox-training/feature-javadoc

Feature javadoc

krcg93 committed:
Change http status and message response

krcg93 committed:
Merge branch 'feature-bookrepository' into feature-bookcrud
